src/java.base/share/classes/com/sun/crypto/provider/AES_Crypt.java

```diff
@@ -0,0 +1,1422 @@
+            0x663BC544, 0x6F36CE4A, 0x7421D358, 0x7D2CD856, 0xA10C7A37,
+            0xA8017139, 0xB3166C2B, 0xBA1B6725, 0x8538560F, 0x8C355D01,
+            0x97224013, 0x9E2F4B1D, 0xE9642247, 0xE0692949, 0xFB7E345B,
+            0xF2733F55, 0xCD500E7F, 0xC45D0571, 0xDF4A1863, 0xD647136D,
+            0x31DCCAD7, 0x38D1C1D9, 0x23C6DCCB, 0x2ACBD7C5, 0x15E8E6EF,
+            0x1CE5EDE1, 0x07F2F0F3, 0x0EFFFBFD, 0x79B492A7, 0x70B999A9,
+            0x6BAE84BB, 0x62A38FB5, 0x5D80BE9F, 0x548DB591, 0x4F9AA883,
+            0x4697A38D,
+    };
+
+    /**
+     * Return the block cipher size.
+     *
+     * @return the AES block cipher size (in bytes).
+     */
+    int getBlockSize() {
+        return AESConstants.AES_BLOCK_SIZE;
+    }
+
+    /**
+     * Verifies if the length argument is a valid key size or not.
+     *
+     * @param len the size of the key (in bytes) to be validated.
+     *
+     * @return {@code true} if the size of the key is valid else {@code false}.
+     */
+    static boolean isKeySizeValid(int len) {
+        return len == AES_128_NKEYS || len == AES_192_NKEYS
+                || len == AES_256_NKEYS;
+    }
+
+    /**
+     * Check algorithm and initalize round keys for decryption and encryption.
+     *
+     * @param decrypting [in] indicates if encrypting ({@code false}) or
+     *      decrypting ({@code true}).
+     * @param algorithm [in] the case insentive string name for the AES cipher.
+     * @param key [in] the symmetric key byte array for encryption/decryption.
+     *
+     * @throws InvalidKeyException if an incorrect algorithm is given or if
+     * an invalid key size is provided.
+     */
+    void init(boolean decrypting, String algorithm, byte[] key)
+            throws InvalidKeyException {
+        int nk;
+        int decrypt = decrypting ? 1 : 0;
+
+        if (!algorithm.equalsIgnoreCase("AES")
+                && !algorithm.equalsIgnoreCase("Rijndael")) {
+            throw new InvalidKeyException ("Invalid algorithm name.");
+        }
+        switch (key.length) {
+            case AES_128_NKEYS -> {
+                rounds = AES_128_ROUNDS;
+                nk = AES_128_NKEYS/WB;
+            }
+            case AES_192_NKEYS -> {
+                rounds = AES_192_ROUNDS;
+                nk = AES_192_NKEYS/WB;
+            }
+            case AES_256_NKEYS -> {
+                rounds = AES_256_ROUNDS;
+                nk = AES_256_NKEYS/WB;
+            }
+            default -> throw new InvalidKeyException(
+                    "Invalid key length (" + key.length + ").");
+        }
+        if (!MessageDigest.isEqual(prevKey, key)) {
+            if (sessionK == null) {
+                sessionK = new int[2][];
+            } else {
+                Arrays.fill(sessionK[0], 0);
+                Arrays.fill(sessionK[1], 0);
+            }
+            if (prevKey != null) {
+                Arrays.fill(prevKey, (byte) 0);
+            }
```

Comment:
Can be moved down to be right before `prevKey = key.clone()` call? This way, `sessionK` assignments are together and not separated by this call